controllers/administration.js

```diff
@@ -1597,12 +1597,29 @@ router.get(
 	},
 );
 
+const currentUserHasPermissionsForRole = (req, res, next) => {
+	const role = req.query.role;
+	const currentUser = res.locals.currentUser;
+
+	let hasPermissions = false;
+
+	if (role === 'student') {
+		hasPermissions = permissionsHelper.userHasPermission(currentUser, ['ADMIN_VIEW', 'STUDENT_LIST'], 'or');
+	} else if (role === 'teacher') {
+		hasPermissions = permissionsHelper.userHasPermission(currentUser, ['ADMIN_VIEW', 'TEACHER_LIST'], 'or');
+	}
```

Comment:
Is it really necessary to distinguish here?

Reply:
yes because this function is used for getting students and teachers as well. And the user who wants to see the teacher list should have TEACHER_LIST permission, and who wants to see students should have STUDENT_LIST permission